EyeOnBuzz/Infrastructure/DataSources/TMDBSession.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  TMDBSession.swift
+//  EyeOnBuzz
+//
+//  Created by Weverton Couto Timoteo on 1/24/18.
+//  Copyright © 2018 Weverton Couto Timoteo. All rights reserved.
+//
+
+import Alamofire
+
+// theMovieDatabaseSession
+class TMDBSession {
+    
+    let httpClient = HTTPClient.init()
+    let notificationCenter = NotificationCenter.default
+    
+    struct Body {
+        static let results = "results"
+
+        struct Pagination {
+            static let page = "page"
+            static let total  = "total_results"
+            static let totalPages = "total_pages"
+        }
+    }
+    
+    // MARK: HTTP Verbs
+    
+    func get(url: URLConvertible, parameters: Parameters?, success: @escaping (Int, Any, Any) -> Void, failure: @escaping (Int, Any) -> Void) {
+        if Connectivity.isNetworkUnreachable() {
+            notificationCenter.post(name: .unreachableNetwork, object: nil)
+        } else {
+            httpClient.get(url: url, parameters: signParameters(parameters)).responseJSON { response in
+                self.handleResponse(response: response, success: success, failure: failure)
+            }
+        }
+    }
+    
+    // MARK: Helpers
+    
+    func signParameters(_ parameters: Parameters?) -> Parameters {
+        var signedParameters = parameters ?? Parameters()
+        
+        signedParameters["api_key"] = Credential.theMovieDatabaseAPIKey()
+        
+        return signedParameters
+    }
+    
+    func handleResponse(response: DataResponse<Any>, success: (Int, [String : AnyObject], Any) -> Void, failure: (Int, Any) -> Void) -> Void {
+        let statusCode: Int! = response.response?.statusCode
```

Comment:
is this 👆 safe?

Reply:
This method will only be called if response (or Alamofire threats) was given, so `statusCode` will always be present as an `Int` value. This approach is not recommended but I would like to hear a suggestion to improve this.

Reply:
It is never safe to assume behavior of external components will not change. They might change and break your code, specially when we consider third party libraries.
The proper way of handling cases like this is using conditional binding like in the snipped bellow:
```Swift
guard let statusCode: Int = response.response?.statusCode else { ... }

Reply:
Yeah, thank you @pomo-mangapinheiro I completely forgot to use `guard` here. About dependencies, I usually write a unit test to define this behavior and freeze dependency version in my Dependency Manager (`carthage` in case).

Reply:
just repasting the comment:



It is never safe to assume behavior of external components will not change. They might change and break your code, specially when we consider third party libraries.
The proper way of handling cases like this is using conditional binding like in the snipped bellow:

```Swift
guard let statusCode: Int = response.response?.statusCode else { ... }

Reply:
Sorry for the mess ;)